pkg/controller/statusmanager/pod_status.go

```diff
@@ -200,6 +203,9 @@ func (status *StatusManager) SetFromPods() {
 		if isNonCritical(dep) && dep.Status.UnavailableReplicas > 0 && !status.installComplete {
 			progressing = append(progressing, fmt.Sprintf("Deployment %q is waiting for other operators to become ready", depName.String()))
 			depProgressing = true
+		} else if dep.Status.UpdatedReplicas < dep.Status.Replicas {
```

Comment:
@kyrtapz  Do we have the same issue with deployment? Is it becacuse the deployments are not running in the management cluster in 4.12 so we didn't see the upgrade issue?

Reply:
I am not sure that we saw the bug with deployments but ovn-k master statefulset would likely be the last thing to update since it has to wait for ovn-k node update. Regardless of that I think deployment update handling should also be fixed in this PR.

Reply:
Yes, makes sense.
I was just to understand if there will be same issue for deployment.